rotkehlchen/chain/ethereum/modules/defisaver/decoder.py

```diff
@@ -64,10 +145,69 @@ def _decode_substorage_action(self, context: DecoderContext) -> DecodingOutput:
 
         return DEFAULT_DECODING_OUTPUT
 
+    def _decode_defisaver_events(self, context: DecoderContext) -> DecodingOutput:
+        """
+        Decodes DeFi Saver events from the given context.
+
+        This method specifically matches the counterparty
+        so to trigger the post-decoding rules.
+        """
+        # We want only to remap the ERC20 transfers, so
+        # we just match the counterparty for triggering the post-decoding rules
+        return DecodingOutput(matched_counterparty=CPT_DEFISAVER)
+
+    def _decode_flashloan(
+            self,
+            decoded_events: list['EvmEvent'],
+            transaction: 'EvmTransaction',
+            all_logs: list['EvmTxReceiptLog'],
+    ) -> list['EvmEvent']:
+        """
+        Decodes flashloan events from the given decoded events.
+
+        This method identifies flashloan ERC20 transfer events originating from DefiSaver flashloan
+        contracts and sets the appropriate event subtype.
+        """
+        for event in decoded_events:
+            if event.address in self.flashloan_addresses and self.base.is_tracked(event.location_label):  # noqa: E501
+                if (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.event_subtype = HistoryEventSubType.GENERATE_DEBT  # TODO: should  we create a new subtype for flashloans?  # noqa: E501
+                    event.counterparty = CPT_DEFISAVER
```

Comment:
Good question. Imo we should have a type for this. The reason is that this will be handled in accounting and the truth is that there was no block where you were in debt. I also saw flashloans withdrawing from palading for example. 

I believe it should be informational/generate debt and informational/payback debt unless we have some reason to mark it as a flashloan.

Maybe @LefterisJP has a better idea/other opinion